### PR TITLE
Tests for tenant restore progress; definition of API types; Fix lints

### PR DESCRIPTION
**Pull request by PowerfooI**
## Summary

* Add test cases for tenant restore and make them pass
* Create API type OBTenantOperation for tenants tiny jobs
* Complete API type definitions according to design draft

* Update `.golangci.yaml`, disable `unused` and `gosimple` linters and `exported`, `struct-tag`, `var-naming`, `comment-spacings` and `unused-receiver` 5 rules of `revive` linter.
* Fix other lint errors
* Install golangci-lint with Makefile target

### CODE REVIEW

File: api/v1alpha1/obtenantoperation_types.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	StandbyTenant string `json:"standbyTenant"`
+}
+
+type OBTenantOpFailoverSpec struct {
+	StandbyTenant string `json:"standbyTenant"`
+}
+
+type OBTenantOpChangePwdSpec struct {
+	Tenant    string                 `json:"tenant"`
+	SecretRef corev1.SecretReference `json:"secretRef"`
```
Comment (chris-sun-star) [inline api/v1alpha1/obtenantoperation_types.go:50]: what will be stored in secret, username and password?

File: api/v1alpha1/obtenantrestore_types.go [hunk 2]
```diff
@@ -25,21 +26,33 @@ type OBTenantRestoreSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	ObClusterName     string `json:"obClusterName"`
-	RestoreTenantName string `json:"restoreTenantName"`
-	Type              string `json:"type"`
-	SourceUri         string `json:"sourceUri"`
-	Until             string `json:"until,omitempty"`
+	TargetTenant string               `json:"targetTenant"`
```
Comment (chris-sun-star) [inline api/v1alpha1/obtenantrestore_types.go:29]: still need obcluster to specify which cluster to restore the tenant

Comment (chris-sun-star) [inline api/v1alpha1/obtenantrestore_types.go:29]: obcluster is needed to specify which cluster the restored tenant located

File: cmd/main.go [hunk 3]
```diff
@@ -48,6 +48,7 @@ func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 
 	utilruntime.Must(v1alpha1.AddToScheme(scheme))
+	utilruntime.Must(oceanbasev1alpha1.AddToScheme(scheme))
```
Comment (chris-sun-star) [inline cmd/main.go:51]: the import is duplicated, oceanbasev1alpha1 and v1alpha1

File: api/v1alpha1/obtenant_types.go [hunk 4]
```diff
@@ -42,6 +45,24 @@ type OBTenantSpec struct {
 	ConnectWhiteList string `json:"connectWhiteList,omitempty"`
 
 	Pools []ResourcePoolSpec `json:"pools"`
+
+	//+kubebuilder:default=PRIMARY
+	TenantRole  constants.TenantRole     `json:"tenantRole,omitempty"`
+	Source      *TenantSourceSpec        `json:"source,omitempty"`
+	Credentials []corev1.SecretReference `json:"credentials,omitempty"`
```
[the remark this answers is not in the document]
Comment (PowerfooI) [reply api/v1alpha1/obtenant_types.go:52]: Maybe it's ok to store users' information in credentials.

File: api/v1alpha1/obtenant_types.go [hunk 5]
```diff
@@ -42,6 +45,24 @@ type OBTenantSpec struct {
...
+
+// Source for restoring or creating standby
+type TenantSourceSpec struct {
+	Tenant  *string        `json:"tenant,omitempty"`
+	Restore *RestoreSource `json:"restore,omitempty"`
+}
+
+type RestoreSource struct {
+	SourceUri      string  `json:"sourceUri"`
+	Until          string  `json:"until"`
```
[the remark this answers is not in the document]
Comment (PowerfooI) [reply api/v1alpha1/obtenant_types.go:63]: Sounds reasonable

File: api/v1alpha1/obtenant_types.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (chris-sun-star) [reply api/v1alpha1/obtenant_types.go:52]: it's ok, but certain users must be specified, like root and standbyro

[the remark this answers is not in the document]
Comment (PowerfooI) [reply api/v1alpha1/obtenant_types.go:52]: Users' credentials in tenant will be implemented on next step.  Some designs are not clear now, maybe user management will be put in OBTenantOperation CR. :)

File: api/v1alpha1/obtenantoperation_types.go (hunk 1 above)
Comment (PowerfooI) [reply api/v1alpha1/obtenantoperation_types.go:50]: Username and password will be here, it's possible to contain other information. Design for tenant credentials are in pending, now they are just placeholders here hahah.

**Review by chris-sun-star — APPROVED** [review]
LGTM

**Pull request merged**